### PR TITLE
Implemented FyCurrencyPipe + Replaced CurrencyPipe in TS

### PR DESCRIPTION
**Pull request by OmkarJ13**

### CODE REVIEW

**Review by abhishek1234321 — APPROVED** [review]

File: src/app/shared/pipes/fy-currency.pipe.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    currencyCode?: string,
+    display?: string | boolean,
+    digitsInfo?: string,
+    locale?: string
+  ): string | null {
+    const transformedValue = this.currencyPipe.transform(value, currencyCode, display, digitsInfo, locale);
+
+    if (transformedValue) {
+      // Gets the index of first digit in the transformed amount string
+      const firstDigitIdx = transformedValue.search(/\d/);
```
Comment (Julias0) [inline src/app/shared/pipes/fy-currency.pipe.ts:28]: what is `Idx`  - dont use abbreviations

File: src/app/shared/pipes/fy-currency.pipe.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+/**
+ * This pipe is a superset of the default CurrencyPipe provided by Angular
+ * It adds functionality to add a space between the currency code and the amount when no symbol is present for the given currency
+ * Since this feature is not there by default, we need to add this by ourselves
+ * For e.g., USD100.00 will be transformed to USD 100.00
+ */
+@Pipe({
+  name: 'currency',
+})
+export class FyCurrencyPipe implements PipeTransform {
```
Comment (Julias0) [inline src/app/shared/pipes/fy-currency.pipe.ts:13]: why not extend the original currency pipe?

File: src/app/shared/pipes/fy-currency.pipe.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    locale?: string
+  ): string | null {
+    const transformedValue = this.currencyPipe.transform(value, currencyCode, display, digitsInfo, locale);
+
+    if (transformedValue) {
+      // Gets the index of first digit in the transformed amount string
+      const firstDigitIdx = transformedValue.search(/\d/);
+      const currencySymbol = transformedValue.substring(0, firstDigitIdx);
+
+      // If the symbol is same as the currency code, we need to add a space for proper readability
```
Comment (Julias0) [inline src/app/shared/pipes/fy-currency.pipe.ts:31]: Add an example please

File: src/app/shared/pipes/fy-currency.pipe.spec.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { CurrencyPipe } from '@angular/common';
+import { FyCurrencyPipe } from './fy-currency.pipe';
+
+describe('FyCurrencyPipe', () => {
```
Comment (Julias0) [inline src/app/shared/pipes/fy-currency.pipe.spec.ts:4]: Please to add unit tests for 100% coverage of this pipe. 
Rest of the pipes all have coverage via unit tests

File: src/app/shared/pipes/fy-currency.pipe.ts (hunk 1 above)
Comment (OmkarJ13) [reply src/app/shared/pipes/fy-currency.pipe.ts:28]: Index 🙈

**Review by Julias0 — APPROVED** [review]

**Pull request merged**